scipy/stats/_variation.py

```diff
@@ -96,33 +99,25 @@ def variation(a, axis=0, nan_policy='propagate', ddof=0, *, keepdims=False):
     """
     xp = array_namespace(a)
     a = xp.asarray(a)
+
     # `nan_policy` and `keepdims` are handled by `_axis_nan_policy`
-    # `axis=None` is only handled for NumPy backend
     if axis is None:
         a = xp.reshape(a, (-1,))
-        axis = 0
-
-    n = a.shape[axis]
-    NaN = _get_nan(a)
-
-    if a.size == 0 or ddof > n:
-        # Handle as a special case to avoid spurious warnings.
-        # The return values, if any, are all nan.
-        shp = list(a.shape)
-        shp.pop(axis)
-        result = xp.full(shp, fill_value=NaN)
-        return result[()] if result.ndim == 0 else result
-
-    mean_a = xp.mean(a, axis=axis)
-
-    if ddof == n:
-        # Another special case.  Result is either inf or nan.
-        std_a = xp.std(a, axis=axis, correction=0)
-        result = xp.where(std_a > 0, xp.copysign(xp.inf, mean_a), NaN)
-        return result[()] if result.ndim == 0 else result
-
-    with np.errstate(divide='ignore', invalid='ignore'):
-        std_a = xp.std(a, axis=axis, correction=ddof)
-        result = std_a / mean_a
+        axis = -1
```

Comment:
No need to change this.
```suggestion
        axis = 0
```